src/lib/bcmp/bcmp.cpp

```diff
@@ -32,11 +34,27 @@
 // 1500 MTU minus ipv6 header
 #define MAX_PAYLOAD_LEN (1500 - sizeof(struct ip6_hdr))
 
+static constexpr uint32_t MESSAGE_TIMER_EXPIRY_PERIOD_MS = 500;
```

Comment:
How did we arrive at this number? It seems to me that it ought to be much smaller. If I imagine for example DFU or lots of video, something with hundreds of requests queued up and waiting half a second for each one once a failure happens, I imagine we might use up all heap and crash before this timeout, or else suddenly everything goes suuuuper sloooow... I don't know for certain, just wondering what our constraints are on this timeout.

Reply:
One constraint is we don't know the RTT of a message N hops away. This is a bit of a goldilocks problem and make sure we pick something big enough for N=5 maybe?

Reply:
If we have log print time stamps we can use that as an upper limit and just multiply that out by the number of N hops

Reply:
Victor, could you please test that and gather some data? Make the longest Bristlemouth chain you easily can with materials at hand. Send a bunch of bcmp cfg get messages as fast as you can from one end of the chain to the other. Gather some stats like min, mean, max, std.

Reply:
on it!

Reply:
Totally not for this PR, but we should probably open a separate characterization ticket, especially when we do DFU - which has a 1024 flash write for each message.

Reply:
For a system of bridge - soft -soft -rbr - normal mote, getting the normal motes configs resulted in:
```
Average difference: 3.5132743362831858 ticks
Standard deviation of differences: 0.5009332490069597 ticks
Minimum difference: 3 ticks
Maximum difference: 4 ticks
```
or about 1 ms per node.

Reply:
will go with 12 ms timer, and 24 ms timeout. This is because we can support up to 16 motes so 16 ms + some padding.

Reply:
After IRL discussion: let's go with 12 ms timer and 24 ms message timeout.